Source/WebCore/Modules/mediastream/gstreamer/GStreamerWebRTCUtils.cpp

```diff
@@ -338,92 +329,134 @@ static String privateKeySerialize(EVP_PKEY* privateKey)
 std::optional<Ref<RTCCertificate>> generateCertificate(Ref<SecurityOrigin>&& origin, const PeerConnectionBackend::CertificateInformation& info)
 {
     ensureDebugCategoryInitialized();
-    EVP_PKEY* privateKey = EVP_PKEY_new();
+    EvpPKeyPtr privateKey;
+
+#if OPENSSL_VERSION_NUMBER < 0x30000000L
+    privateKey.reset(EVP_PKEY_new());
     if (!privateKey) {
         GST_WARNING("Failed to create private key");
         return { };
     }
-
-    auto scopeExit = WTF::makeScopeExit([&] {
-        EVP_PKEY_free(privateKey);
-    });
+#endif
 
     switch (info.type) {
     case PeerConnectionBackend::CertificateInformation::Type::ECDSAP256: {
-        EC_KEY* ecKey = EC_KEY_new_by_curve_name(NID_X9_62_prime256v1);
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
```

Comment:
Are these checks needed when OpenSSL 3 is now demanded at configure-time?

Reply:
Not anymore, but I thought I'd get rid of them in a follow-up patch, as noted in commit message.

Reply:
Understood.